pkg/executables/docker.go

```diff
@@ -89,7 +95,9 @@ func (d *Docker) TagImage(ctx context.Context, image string, endpoint string) er
 }
 
 func (d *Docker) PushImage(ctx context.Context, image string, endpoint string) error {
-	localImage := strings.ReplaceAll(image, defaultRegistry, endpoint)
+	replacer := strings.NewReplacer(defaultRegistry, endpoint, packageProdLocation, endpoint, packageDevLocation, endpoint)
```

Comment:
same comment about doing this in the registry mover

Reply:
Added an issue for this: https://github.com/aws/eks-anywhere/issues/3206